### PR TITLE
fix: incorrect instructions for site documentation

### PR DESCRIPTION
**Pull request by reginaelyse**
I noticed the docs for how to build and serve documentation locally were wrong, so making this PR to correct it. 
The make target for lint-docs was removed in https://github.com/argoproj/argo-cd/pull/8722 but was never removed from documentation. 

Proposed new changes look like:
Screenshot from 2023-04-12 16-20-45

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.

Please see [Contribution FAQs](https://argo-cd.readthedocs.io/en/latest/developer-guide/faq/) if you have questions about your pull-request.

### CODE REVIEW

File: docs/developer-guide/site.md [hunk 1]
```diff
@@ -9,11 +9,12 @@ To test:
...
 make serve-docs
 ```
+Once running, you can view your locally built documentation at [http://0.0.0.0:8000/](http://0.0.0.0:8000/). 
 
 Check for broken external links:
 
 ```bash
-make lint-docs
+make build-docs
 ```
```
Comment (crenshaw-dev) [inline docs/developer-guide/site.md:18]: I think we can just drop this whole section. `build-docs` doesn't actually check for broken links. And there's really no need to run `build-docs` if everything looks okay after `serve-docs`.

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

**Review by ishitasequeira — APPROVED** [review]

**Review by crenshaw-dev — APPROVED** [review]
Thanks, @reginapizza!

**Pull request merged**